camel/o1datagen/o1datagen.py

```diff
@@ -0,0 +1,332 @@
+        self.golden_answers = golden_answers
+        self.search_limit = search_limit
+        self.solution_tree: Dict[str, Dict] = defaultdict(dict)
+        logger.info(
+            "O1DataGene initialized with search_limit=%d", search_limit
+        )
+
+    def get_answer(self, question: str, context: str = "") -> str:
+        r"""Get the AI's thought process and answer.
+        Args:
+            question (str): The problem or question to be solved by the AI.
+            context (str, optional): Additional context or existing content to
+            consider when generating the answer. Defaults to an empty string.
+        Returns:
+            str: The AI's detailed response including
+            thought process and final answer."""
+        prompt = f"""
+        Please think step by step and solve this problem: {question}
+        Existing content: {context}
+        Requirements:
+        1. Analyze the problem requirements
+        2. List the steps to solve the problem
+        3. Execute the solution process
+        4. Provide the final answer
+        Please explain the thought process of each step in detail.
+        """
+        response = self.chat_agent.step(prompt)
+        answer = response.msgs[0].content
+        logger.info("AI thought process:\n%s", answer)
+        return answer
+
+    def verify_answer(self, question: str, answer: str) -> bool:
+        r"""Verify if a generated answer is semantically equivalent to
+        the golden answer for a given question.
+        Args:question (str): The question to look up
+        in the golden answers dictionary.
+              answer (str): The answer to verify, typically generated
+              by a model or provided by a user.
+        Returns:bool: True if the answer matches the golden answer
+        based on semantic equivalence
+              (meaning the core content and meaning are the same,
+              even if the exact wording differs).
+              False in the following cases:
+              - If the golden answers dictionary is empty
+              - If the provided question doesn't exist in the golden answers
+              - If the answer's meaning differs from the golden answer
+        """
+        if not self.golden_answers:
+            logger.error(
+                """golden_answers is empty.
+                 Please import QA data before verification."""
+            )
+            return False
+        golden_answer = self.golden_answers[question]
+        if not golden_answer:
+            raise ValueError(
+                f"""Golden answer for question '{question}' is empty.
+                Cannot proceed with verification."""
+            )
+        prompt = f"""Please determine if the following two answers 
+        express the same meaning:
+        Question: {question}
+        Answer 1: {answer}
+        Answer 2: {golden_answer}
+        Just answer "True" or "False".
+        """
+        response = self.chat_agent.step(prompt)
```

Comment:
use `self.chat_agent.reset()` to clear memory before step new message, same as below agent.step

Reply:
thanks already add it